.github/workflows/integration.yml

```diff
@@ -0,0 +1,101 @@
+name: 'integration-test'
+on:
+  workflow_run:
+    workflows: [build]
+    types:
+      - completed
+
+env:
+  docker_tag: '${{ github.sha }}'
+  docker_repo: 'us-docker.pkg.dev/lumberjack-dev-infra/images'
+  wif_provider: 'projects/638387980668/locations/global/workloadIdentityPools/github-pool/providers/github-provider'
+  wif_service_account: 'gh-access-sa@lumberjack-dev-infra.iam.gserviceaccount.com'
+  server_project_id: 'github-ci-server'
+  client_project_id: 'github-ci-app-0'
+  bigquery_dataset_id: 'audit_logs.audit_abcxyz_data_access'
+
+jobs:
+
+  deploy-infra:
+    runs-on: 'ubuntu-latest'
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    env:
+      tf_module_dir: './terraform/modules/ci'
+    outputs:
+      http_client_endpoints: ${{ steps.tf_apply.outputs.http_client_endpoints }}
+      grpc_client_endpoints: ${{ steps.tf_apply.outputs.grpc_client_endpoints }}
+    steps:
+      - name: 'Checkout'
+        uses: 'actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b' # ratchet:actions/checkout@v3
+      - name: 'Setup Terraform'
+        uses: hashicorp/setup-terraform@17d4c9b8043b238f6f35641cdd8433da1e6f3867 # ratchet:hashicorp/setup-terraform@v2
+        with:
+          terraform_wrapper: false
+      - name: 'Init the terraform infrastructure'
+        run: terraform -chdir=${{ env.tf_module_dir }} init
+      - name: 'Apply the infrastructure updates'
+        id: 'tf_apply'
+        env:
+          client_images: '{\"ljc-go-grpc\":\"${{env.docker_repo}}/ljc-go-grpc:${{env.docker_tag}}\",\"ljc-go-http\":\"${{env.docker_repo}}/ljc-go-http:${{env.docker_tag}}\",\"ljc-java-grpc\":\"${{env.docker_repo}}/ljc-java-grpc:${{env.docker_tag}}\",\"ljc-java-http\":\"${{env.docker_repo}}/ljc-java-http:${{env.docker_tag}}\"}'
+          server_image: '${{env.docker_repo}}/lumberjack-server:${{env.docker_tag}}'
+        run: |-
+          terraform -chdir=${{ env.tf_module_dir}} apply -auto-approve \
+            -var="server_project_id=${{ env.server_project_id }}" \
+            -var="server_image=${{ env.server_image }}" \
+            -var="server_service_name=lumberjack-server" \
+            -var="client_project_id=${{ env.client_project_id }}" \
+            -var="client_images=${{ env.client_images }}" 
+          terraform -chdir=${{ env.tf_module_dir}} output -json client_endpoints | jq -c > output.json
+
+          grpc_client_endpoints=$(jq -c '{"ljc-go-grpc","ljc-java-grpc"} | to_entries | map(.value|tostring)' < output.json)
+          http_client_endpoints=$(jq -c '{"ljc-go-http","ljc-java-http"} | to_entries | map(.value|tostring)' < output.json)
+
+          echo "::set-output name=grpc_client_endpoints::$(echo ${grpc_client_endpoints})"
```

Comment:
This is super nitty and we can do this in a follow-up PR, but I notice we're setting GitHub Actions outputs and then later on L94 we're converting them to envvars. I think it could be cleaner if we just exported these directly into `GITHUB_ENV`:

```text
echo "GRPC_ENDPOINTS=${grpc_client_endpoints}" >> $GITHUB_ENV
```